src/connector/src/source/cdc/source/reader.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures::{pin_mut, StreamExt};
+use futures_async_stream::try_stream;
+use risingwave_common::util::addr::HostAddr;
+use risingwave_pb::cdc_service::{DbConnectorProperties, GetEventStreamResponse};
+use risingwave_rpc_client::CdcClient;
+
+use crate::source::base::{SourceMessage, SplitReader};
+use crate::source::cdc::CdcProperties;
+use crate::source::{BoxSourceStream, Column, ConnectorState, SplitImpl};
+
+pub struct CdcSplitReader {
+    source_id: u64,
+    props: CdcProperties,
+}
+
+#[async_trait]
+impl SplitReader for CdcSplitReader {
+    type Properties = CdcProperties;
+
+    async fn new(
+        props: CdcProperties,
+        state: ConnectorState,
+        _columns: Option<Vec<Column>>,
+    ) -> Result<Self> {
+        if let Some(splits) = state {
+            // expect only one split
+            debug_assert!(splits.len() == 1);
+            for split in &splits {
+                if let SplitImpl::Cdc(cdc_split) = split {
```

Comment:
```suggestion
        let cdc_split = state.exactly_one().map_err(...)?;
```

Reply:
Sorry, I do not find this `exactly_one` function in `state` or `splits`. something wrong?

Reply:
you may need `use itertools` for getting the element out of the vec
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.exactly_one

Reply:
Got it. Thanks